src/main/java/mate/academy/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package mate.academy.bookstore.service.impl;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.bookstore.dto.shoppingcart.cartitem.CartItemRequestDto;
+import mate.academy.bookstore.dto.shoppingcart.cartitem.CartItemResponseDto;
+import mate.academy.bookstore.dto.shoppingcart.cartitem.CartItemUpdateDataRequestDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.CartItemMapper;
+import mate.academy.bookstore.mapper.ShoppingCartMapper;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.model.CartItem;
+import mate.academy.bookstore.model.ShoppingCart;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.book.BookRepository;
+import mate.academy.bookstore.repository.shoppingcart.CartItemRepository;
+import mate.academy.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.bookstore.service.ShoppingCartService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto findShoppingCart(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> {
+                    ShoppingCart newShoppingCart = new ShoppingCart();
+                    newShoppingCart.setUser(user);
+                    return shoppingCartRepository.save(newShoppingCart);
+                });
```

Comment:
I still don't like idea where we put something in DB during GET operation flow
```suggestion
        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
                .orElseThrow(() -> new IllegalStateException("Cart cannot be found for..."));
```